### PR TITLE
Added Emotion Events and Player.Emotion

### PR DESCRIPTION
**Pull request by Rysik5318**
## Description
**Describe the changes** 

**What is the current behavior?** None

**What is the new behavior?** 
Adds the ChangingEmotion and ChangedEmotion events.
Adds Emontion variable to the Player class.

**Other information**:

<br />

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [x] I have checked the project can be compiled
- [x] I have tested my changes and it worked as expected

### Other
- [ ] Still requires more testing

### CODE REVIEW

File: EXILED/Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -732,6 +733,18 @@ public bool IsBypassModeEnabled
...
+        /// <summary>
+        /// Gets or sets the player's emotion.
+        /// </summary>
+        public EmotionPresetType Emotion
+        {
+            get => EmotionSync.GetEmotionPreset(ReferenceHub);
+            set
+            {
+                EmotionSync.ServerSetEmotionPreset(ReferenceHub, value);
+            }
```
Comment (VALERA771) [inline EXILED/Exiled.API/Features/Player.cs:745]:
```suggestion
            set => EmotionSync.ServerSetEmotionPreset(ReferenceHub, value);
```

**Review by VALERA771 — APPROVED** [review]
I'll pretend I didn't see bool prefix

**Review by louis1706 — APPROVED** [review]

File: EXILED/Exiled.Events/Handlers/Player.cs [hunk 2]
```diff
@@ -549,6 +549,28 @@ public class Player
...
+
+        /// <summary>
+        /// Invoked after a player's emotion changed.
+        /// </summary>
+        public static Event<ChangedEmotionEventArgs> ChangedEmotion { get; set; } = new();
+
+        /// <summary>
+        /// Called before a player's emotion changed.
+        /// </summary>
+        /// <param name="ev">The <see cref="ChangingAmmoEventArgs"/> instance.</param>
```
Comment (louis1706) [inline EXILED/Exiled.Events/Handlers/Player.cs:565]:
```suggestion
        /// <param name="ev">The <see cref="ChangingEmotionEventArgs"/> instance.</param>
```

**Review by IRacle1 — CHANGES_REQUESTED** [review]
cool!!!!!!!!

Now do this without bool prefix and postfix😈

**Comment by louis1706** [comment]
> cool!!!!!!!!
> 
> Now do this without bool prefix and postfix😈

actually Postfix is fine just bool prefix is not

**Review by VALERA771 — COMMENTED** [review]
Can you add new emoji preset setter support to transpiler?

**Comment by Misaka-ZeroTwo** [comment]
https://github.com/Rysik5318/EXILED/pull/1/

**Comment by Rysik5318** [comment]
> Can you add new emoji preset setter support to transpiler?

I don't know how to make a transpiller

**Review by VALERA771 — APPROVED** [review]

**Pull request merged**